src/controllers/photoController.ts

```diff
@@ -0,0 +1,91 @@
+import { NextFunction, Request, Response } from "express";
+import { deletePhoto, uploadPhoto } from "../services/photoService";
+import User, { IUser } from "../models/User";
+import { NotFoundError, UnauthenticatedError } from "../errors";
+
+export const uploadPhotoController = async (
+  req: Request,
+  res: Response, 
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+    if (!req.file) {
+      res.status(400).json({ message: "No file uploaded." });
+      return;
+    }
+
+    const { uploadResult, optimizeUrl, autoCropUrl } = await uploadPhoto(
+      req.file
+    );
+
+    const userProfile = await User.findOne({ "_id": user.userId});
+        
+    if (!userProfile) {
+      return next(new NotFoundError("The user was not found"));
+    }
+
+    userProfile.photoId = uploadResult.public_id;
+    userProfile.photo = optimizeUrl;
+    await userProfile.save();
+
+
+    res.json({
```

Comment:
I assume this defaults to a 200 status, but I think it would be nice to explicitly give a 200 status here (and in the delete route too).